addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/meter/DSMRMeter.java

```diff
@@ -0,0 +1,151 @@
+     */
+    public DSMRMeter(DSMRMeterDescriptor meterDescriptor, DSMRMeterListener meterListener) {
+        if (meterListener == null) {
+            throw new IllegalArgumentException("meterListener can not be null");
+        }
+        this.meterDescriptor = meterDescriptor;
+        this.meterListener = meterListener;
+
+        supportedIdentifiers = new LinkedList<>();
+
+        for (CosemObjectType msgType : meterDescriptor.getMeterType().supportedCosemObjects) {
+            OBISIdentifier obisId = msgType.obisId;
+            if (msgType.obisId.getGroupB() == null) {
+                supportedIdentifiers.add(new OBISIdentifier(obisId.getGroupA(), meterDescriptor.getChannel(),
+                        obisId.getGroupC(), obisId.getGroupD(), obisId.getGroupE(), msgType.obisId.getGroupF()));
+            } else {
+                supportedIdentifiers.add(msgType.obisId);
+            }
+        }
+    }
+
+    /**
+     * Returns a list of Cosem Objects this meter will handle
+     *
+     * @param cosemObjects the list of CosemObject that must be processed
+     * @return List of CosemObject that this meter can process
+     */
+    private List<CosemObject> filterMeterValues(List<CosemObject> cosemObjects) {
+        List<CosemObject> filteredList = new LinkedList<>();
+
+        logger.trace("supported identifiers: {}, searching for objects {}", supportedIdentifiers, cosemObjects);
+        for (CosemObject cosemObject : cosemObjects) {
+            if (supportedIdentifiers.contains(cosemObject.getObisIdentifier().getReducedOBISIdentifier())) {
+                logger.trace("Added supported cosemObject {}", cosemObject);
+                filteredList.add(cosemObject);
+            }
+        }
+        return filteredList;
+    }
+
+    /**
+     * Handles the new CosemObjects
+     *
+     * This method will process the messages specific for this DSMRMeter
+     *
+     * @param cosemObjects List of CosemObject to process
+     *
+     * @return List of OBISMessages that were processed by this DSMRMeter
+     */
+    public List<CosemObject> handleCosemObjects(List<CosemObject> cosemObjects) {
+        List<CosemObject> filteredValues = filterMeterValues(cosemObjects);
+
+        for (CosemObject newValue : filteredValues) {
+            logger.trace("Send new meter value {}", newValue);
+            meterListener.meterValueReceived(newValue);
+        }
+        return filteredValues;
+    }
+
+    /**
+     * Returns the Meter identification {@link DSMRMeterIdentification}
+     *
+     * @return the identification of this meter
+     */
+    public DSMRMeterDescriptor getMeterDescriptor() {
+        return meterDescriptor;
+    }
+
+    /**
+     * Returns true if the identification and metertype are equal
+     *
+     * @return true if the identification and metertype are equal
+     */
+    @Override
+    public boolean equals(Object other) {
```

Comment:
With equals you should always implement hashcode, please check all occurrences of either equals and hashcode.